code/iff_defs/iff_defs.cpp

```diff
@@ -289,30 +399,56 @@ void iff_init()
 		// begin reading data
 		while (required_string_either("#End", "$IFF Name:"))
 		{
-			iff_info *iff;
-			int cur_iff;
-
-			// add new IFF
-			cur_iff = (int) Iff_info.size();
-			Iff_info.push_back(iff_info());
-			attack_names.emplace_back();
-			observed_color_table.emplace_back();
-			iff = &Iff_info[cur_iff];
-
-
-			// get required IFF info ----------------------------------------------
 
 			// get the iff name
 			required_string("$IFF Name:");
-			stuff_string(iff->iff_name, F_NAME, NAME_LENGTH);
+			stuff_string(ifft.iff_name, F_NAME, NAME_LENGTH);
+			mprintf(("IFFs got to name %s\n", ifft.iff_name));
+			if (optional_string("+nocreate")) {
+				if (!Parsing_modular_table) {
+					Warning(LOCATION, "+nocreate flag used for iff in non-modular table\n");
```

Comment:
Since we are in table parse, use ``error_display(0, "Message");`` for warnings over ``Warning``, as this'll log the exact line the warning came from to the user